pkg/gatewayserver/io/udp/udp.go

```diff
@@ -328,7 +328,8 @@ func (s *srv) handleUp(ctx context.Context, state *state, packet encoding.Packet
 			}
 		}
 		var rtt *time.Duration
-		if downlink, delta, ok := state.tokens.Get(uint16(packet.Token[0])<<8|uint16(packet.Token[1]), packet.ReceivedAt); ok {
+		downlink, delta, ok := state.tokens.Get(uint16(packet.Token[0])<<8|uint16(packet.Token[1]), packet.ReceivedAt)
+		if ok && downlink != nil {
 			msg.TxAcknowledgment.DownlinkMessage = downlink
 			msg.TxAcknowledgment.CorrelationIDs = downlink.CorrelationIDs
 			rtt = &delta
```

Comment:
I don't see a case where there's no downlink but a corresponding time stamp is stored to check the TxAck against.